lib/notifications/display.dart

```diff
@@ -502,7 +503,19 @@ class NotificationDisplayManager {
     final route = routeForNotification(context: context, url: url);
     if (route == null) return; // TODO(log)
 
-    // TODO(nav): Better interact with existing nav stack on notif open
+    int? currentAccountId;
+    navigator.popUntil((r) {
+      if (r is AccountRoute) {
+        currentAccountId = r.accountId;
+      }
+      return true;
+    });
+
+    if (currentAccountId != route.accountId) {
```

Comment:
When `currentAccountId` (i.e., account ID associated with the top-most route) is `null`, routes below the top-most route can still belong to the same account. `HomePage.navigate` will pop those routes even though they belong to same account that receives this notification.